src/continuous_agg.c

```diff
@@ -444,80 +475,61 @@ drop_continuous_agg(ContinuousAgg *agg, bool drop_user_view)
 		ts_bgw_job_delete_by_id(job->fd.id);
 	}
 
-	user_view = (ObjectAddress){
-		.classId = RelationRelationId,
-		.objectId =
-			get_relname_relid(NameStr(agg->data.user_view_name),
-							  get_namespace_oid(NameStr(agg->data.user_view_schema), false)),
-	};
-	/* The partial view may already be dropped by PG's dependency system (e.g. the raw table was
-	 * dropped) */
-	if (OidIsValid(user_view.objectId))
-		LockRelationOid(user_view.objectId, AccessExclusiveLock);
-
-	raw_hypertable = ts_hypertable_get_by_id(agg->data.raw_hypertable_id);
-	/* The raw hypertable might be already dropped if this is a cascade from that drop */
```

Comment:
These comments  about locking and situtaions related to dependent objects being dropped, should be retained. Otherwise, it is not clear why it is ok to not throw an error if the object is not found.

Reply:
I've updated the function description in order to describe the situation. There were couple of duplicate messages about the locking, not sure how much value they have. I can put them back, if you would like

Reply:
I think it is still good to keep these comments about locking and  objects that could have been potentially dropped (line 459 , 463 , 464 and 500)..